### PR TITLE
Add OstPools

### PR DESCRIPTION
**Pull request by utopiabound**
Add structure of ost pools, able to add/remove pools
Add/remove osts to/from pools

Issue #1241

### CODE REVIEW

File: chroma_core/models/filesystem.py [hunk 1]
```diff
@@ -97,6 +97,9 @@ def get_servers(self):
         # (http://stackoverflow.com/questions/4764110/django-template-cant-loop-defaultdict)
         return dict(servers)
 
+    def get_pools(self):
+        return OstPool.objects.filter(Q(filesystem=self))
```
Comment (jgrund) [inline chroma_core/models/filesystem.py]: What sql does this create? Can (should) we be more specific?

File: chroma_core/models/filesystem.py [hunk 2]
```diff
@@ -406,8 +409,155 @@ def description(self):
...
+    def can_run(self, instance):
+        mgs = instance.filesystem.mgs
+        return mgs is not None and mgs.active_host is not None
+
+    @classmethod
+    def get_args(cls, pool):
+        return {"pool_id": pool.id}
+
+    def description(self):
+        return "Initiate a reboot on host %s" % self.host
```
Comment (jgrund) [inline chroma_core/models/filesystem.py]: Update description.

File: iml-agent/src/action_plugins/ostpool.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        ost.as_str(),
+    ])
+    .map(drop)
+}
+
+pub fn pool_destroy(
+    filesystem: String,
+    name: String,
+) -> impl Future<Item = (), Error = ImlAgentError> {
+    lctl(&["pool_destroy", format!("{}.{}", filesystem, name).as_str()]).map(drop)
```
Comment (jgrund) [inline iml-agent/src/action_plugins/ostpool.rs]:
```suggestion
    lctl(&["pool_destroy", &format!("{}.{}", filesystem, name)]).map(drop)
```

File: iml-agent/src/action_plugins/ostpool.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+pub fn pools(filesystem: String) -> impl Future<Item = Vec<OstPool>, Error = ImlAgentError> {
+    lctl(&["pool_list", filesystem.as_str()]).map(|o| {
```
Comment (jgrund) [inline iml-agent/src/action_plugins/ostpool.rs]:
```suggestion
    lctl(&["pool_list", &filesystem]).map(|o| {
```

**Comment by utopiabound** [comment]
Fun and excitement with OST Pools and MGS that are separate from MDT0:
* Prior to Lustre 2.12.3 - An MGS mounted on one node and MDT0 mounted on the other node, no pool commands would work on the MGS, thus there was no way alter or create pools.
* Lustre 2.12.3 - Altering or creating pools works from MGS (and only MGS), but pool_list only works from a client or MDT.

File: chroma_core/models/filesystem.py (hunk 1 above)
Comment (utopiabound) [reply chroma_core/models/filesystem.py]: This should be returning all pools for the given filesystem.

Comment (johnsonw) [inline chroma_core/models/filesystem.py]: Is the `Q` necessary? There are many places where we filter without the Q.

File: iml-manager-cli/src/ostpool.rs [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                vec![
+                    ("name", poolname.as_str()),
+                    ("filesystem", fs.id.to_string().as_str()),
+                ],
+            )
+            .await?;
+            term.write_line(&format!("{} ost pool...", style("Destroying").green()))?;
+            let objs: ObjCommands = resp.json().await?;
+            wait_for_cmds(objs.commands).await?;
+            //println!("HEADERS:\n{:?}", resp.headers());
```
Comment (johnsonw) [inline iml-manager-cli/src/ostpool.rs]: dead code?

File: chroma_core/models/filesystem.py (hunk 1 above)
Comment (utopiabound) [reply chroma_core/models/filesystem.py]: I'm not sure.  I just copied a filter line from somewhere else, I'll remove the Q()

**Comment by utopiabound** [comment]
This still needs
* [ ] Action Daemon to detect existing/changed OST Pools
* [ ] ostpool API Validation
* [ ] Error checking prior to failed `lctl pool_add`

File: iml-agent/src/action_plugins/ostpool.rs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+pub async fn pool_create(filesystem: String, name: String) -> Result<(), ImlAgentError> {
+    let pn = format!("{}.{}", filesystem, name);
+    lctl(vec!["pool_new", &pn]).await.map(drop)
+}
+
+/// This needs to be a seperate action from pool_create() since pool create runs on MGS
+/// and this runs on MDS
+pub async fn action_pool_wait(cmd: CmdPool) -> Result<(), ImlAgentError> {
+    let time_to_wait = 120;
+    // wait up to a 2 minutes
```
Comment (jgrund) [inline iml-agent/src/action_plugins/ostpool.rs]:
```suggestion
    // wait up to 2 minutes
```

File: liblustreapi-types/build.rs [hunk 7]
```diff
@@ -26,6 +26,7 @@ fn main() {
         .whitelist_type("lov_user_mds_data_v1")
         .whitelist_type("fid_array")
         .whitelist_var("OBD_MAX_FIDS_IN_ARRAY")
+        .whitelist_var("LUSTRE_MAXFSNAME")
```
Comment (jgrund) [inline liblustreapi-types/build.rs]: How likely is this to ever change?

**Pull request merged**